Document --data-values-file now accepts directories

pivotaljohn opened the pull request.

jtigger committed:
Document --data-values-file now accepts directories

also:
- and that --data-value-file accepts URLs and standard in too
- surface each `--data-value...` flag as a header to improve SEO
- include explicit references to examples from the codebase and blog
  articles to make these resources more visible
- beef-up a couple examples to illustrate more what's possible
- ensure all examples include output to make explicit the feature's
  effect

gcheadle-vmware commented on site/content/ytt/docs/develop/ytt-data-values.md:
````diff
@@ -21,110 +21,239 @@ Data Values can be configured in one of two ways:
+Given https://raw.githubusercontent.com/vmware-tanzu/carvel-ytt/develop/examples/data-values/values-file.yml
+
+```console
+ytt --data-values-file https://raw.githubusercontent.com/vmware-tanzu/carvel-ytt/develop/examples/data-values/values-file.yml --data-values-inspect
````
Add `$` for console

gcheadle-vmware commented on site/content/ytt/docs/develop/ytt-data-values.md:
```diff
@@ -21,110 +21,239 @@ Data Values can be configured in one of two ways:
 - `PREFIX` — the literal prefix used to select the set of environment variables from which to configure Data Values.
 - `@lib:` — (optional) specify library whose data values to configure (details [below](#setting-library-values-via-command-line-flags))
+
+**Setting Environment Variables**
```
I didn't know this, interesting!

gcheadle-vmware reviewed (approved):
All around a really positive change for the data values + schema documentation. I left some comments about some small changes I thought should be done. Otherwise, it looks great!

jtigger committed:
Update "How it Works" with Data Values Files

- update diagram to include Data Values Files, visual conveying that
  this is a separate input.
- increase usefulness of table of contents by promoting the four
  pipelines steps to H3s.
- improve navigability of the page by listing document types with links
  to anchors.

pivotaljohn commented on site/content/ytt/docs/develop/how-to-use-data-values.md:
```diff
@@ -4,102 +4,137 @@ title: Using Data Values
-  load_balancer:
-    enabled: true
-    external_ip: 172.120.12.232
+ytt -f schema.yml -f config.yml --data-values-file values.yml
```
No, good catch.

I've started removing the leading `$ ` in cases where the code fence contains _only_ the command-line and it's something the user can execute (or nearly) as is.

pivotaljohn replied:
Oh, no, I see that it's GitHub, not Hugo that, when the reader mouses over the code fence, that they include a button to copy the contents.  I'll be putting the `$ ` in all these cases.

pivotaljohn commented on site/content/ytt/docs/develop/how-to-use-data-values.md:
```diff
@@ -4,102 +4,137 @@ title: Using Data Values
+- Setting a value for an _array_ in schema: \
   https://github.com/vmware-tanzu/carvel-ytt/tree/develop/examples/schema-arrays
+- Using most of the `--data-value...` flags:\
+  https://github.com/vmware-tanzu/carvel-ytt/tree/develop/examples/data-values/
```
Yeah, I noted that... I sensed I had expanded the scope quite a bit from the original intent and figure while this wasn't as nice as having a README.md, it was "good enough"... I'd like to see a README.md for _each_ example, giving more context.

pivotaljohn commented on site/content/ytt/docs/develop/ytt-data-values.md:
````diff
@@ -21,110 +21,239 @@ Data Values can be configured in one of two ways:
+key4: <contents of client.crt>
+```
+
+**Notes**
````
Thank you!

pivotaljohn commented on site/content/ytt/docs/develop/how-to-use-data-values.md:
```diff
@@ -4,102 +4,137 @@ title: Using Data Values
+- Data Values Overlay(s) through the `--file` flag
 
-_(For a high-level overview of `ytt`, see [How it works](how-it-works.md).)_
+This guide illustrates this flow.
```
Thank you. I had writer's fatigue and failed to circle back, here.

jtigger committed:
Improve Data Values docs based on feedback

jtigger committed:
Tweak names of things to be more concrete

- "Customizations" ==> "Custom Values"

Also:
- added hugo_build.lock to .gitignore

jtigger committed:
More wordsmithing
